scripts/gpg-share.sh

```diff
@@ -85,29 +78,19 @@ check_file () {
 }
 
 
-file="$1"; shift 1
-check_file "$file"
-
-if [ -n "$USER_FILE" ]; then
-  check_file "$USER_FILE"
-  readarray users < "$USER_FILE"
-else
-  users=($@)
-fi
-
-
 # TODO: documentation
 check_users() {
+  users=("$@")
   [ "${#users[@]}" -eq 0 ] && {
     log_error "Please give at least one gpg id after the first argument"
+    usage
     exit 1
   }
 
   # check if each of the "user" is in the keyring
   invalid=0
-  for _user in "${users[@]}";
+  for user in "${users[@]}";
   do
-      user=$(echo -n "$_user")
       nb_ids=$(gpg --quiet --list-keys "$user" 2> /dev/null | grep "^uid" | wc -l)
```

Comment:
Actually I just discovered that you should use `--with-colons` here
> Avoid using the output of this command in scripts or other programs as it is likely to change as GnuPG changes. See --with-colons for a machine-parseable key listing command that is appropriate for use in scripts and other programs. 

From [gpg man page](https://manpages.org/gpg)

Reply:
actually, this is super interesting :open_mouth: :smirk: 

however, i simply want to count the numbers of keys with the same user name as the input :thinking: 
and as each key entry has a unique "_uid_" at the beginning of the line, the other one being "_pub_", "_sig_", "_sig 3_" and "_sub_"

in the end, i simply do not see where this will improve the count and make the script "safe" :yum:

Reply:
That's right actually it won't (at least should not) make the script safer!

But I wanted to point this out to share my discovery **and** because even if it doesn't improve the script: it's still good practice :yum: 
(example: someone else wants to modify the script and doesn't know this flag, if you already use it he is more likely to use it too! making the script better :wink: )

Reply:
i love what you said.
so i will use `gpg --with-colons` for this :muscle:

Reply:
should be good to go in 67a797e :yum: 

i've still been able to encrypt the `README` into an archive, share it to myself (lol), un`tar` it, decrypt  it with `gpg` and read the result :ok_hand: 
and the script checks still look to work :+1: